lib/Doctrine/ORM/Tools/SchemaValidator.php

```diff
@@ -255,6 +256,12 @@ public function validateClass(ClassMetadataInfo $class)
             }
         }
 
+        if (! $class->isInheritanceTypeNone() && ! $class->isRootEntity() && array_search($class->name, $class->discriminatorMap) === false) {
```

Comment:
Hello @beberlei,

shouldn't we skip this for abstract classes?
They were allowed to not be a part of mapping, but just be parent class for classes that are mapped in the root entity.

Now with update to Doctrine version 2.9.1 we are getting this error by schema validator
```
Entity class 'App\Entity\Order\CardSuborder' is part of inheritance hierarchy, but is not mapped in the root entity 'App\Entity\Suborder' discriminator map. All subclasses must be listed in the discriminator map.
```

where `Suborder` is a root entity, and `CardSuborder` is an abstract class extended from it, and then we have other not abstract classes extended from `CardSuborder`, and mapped in `Suborder`.

Reply:
Same issue for me in #8771; any solution for this, @beberlei?